src/data/nat/choose/central.lean

```diff
@@ -126,7 +142,16 @@ begin
 end
 
 /--
-Sufficiently large primes appear only to multiplicity 0 or 1 in the central binomial coefficient.
+A `pow` form of `nat.padic_val_nat_central_binom_le`
+-/
+lemma pow_padic_val_nat_central_binom_le_two_mul {p n : ℕ} (hp : p.prime) (n_pos : 0 < n) :
+  p ^ (padic_val_nat p (central_binom n)) ≤ 2 * n :=
```

Comment:
Do we really want to be using `padic_val_nat` to talk about this, rather than `nat.factorization`? (cc @ericrbg)

Reply:
Probably not, but frankly I want to have a conversation about `padic_val_nat` anyway. It's strange that `padic_val_nat` can be used for `p` that aren't prime.

Reply:
That's just standard mathlib design. Separate propositions from data unless strictly needed.

Reply:
I'm pretty sure the intended pattern is to use `nat.factorization` rather than any of the other spellings, but I don't know if that's feasible just yet

Reply:
these theorems seem to be proved with `multiplicity` (c.f. [`nat.prime.multiplicity_choose`](https://leanprover-community.github.io/mathlib_docs/data/nat/multiplicity.html#nat.prime.multiplicity_choose)) and the `padic_val_nat` is rewritten instantly so I definitely think this could be done using `multiplicity`

Reply:
we don't seem to have that for `p` prime `multiplicity = factorization`, I think that's probably a relevant result

Reply:
oh we do, `multiplicity_eq_factorization`, it was just in a different file than I thought

Reply:
I mean, it is the p-adic valuation on the right values

Reply:
To circle back to the original suggestion, I agree `nat.factorization` would be better, but several preexisting lemmas in the file use `padic_val_nat`. I'm inclined to say this should be left for another PR.

Reply:
I very strongly disagree with this suggestion. Redefining `padic_val_nat` so that `p` had to be prime would do nothing but make it more cumbersome to use. You don't need the hypothesis in the definition, so why require it? You might even be able to prove a few results in the case where `p` is not prime, which makes it so that you need to carry around the hypothesis less than you would otherwise.

In any case, I don't see how this design could ever make things harder for you. Most of your theorems probably have an assumption that `p` is prime anyways, so what does it matter that `padic_val_nat` is defined for values outside of it?

Reply:
Yes, I think I've had and lost this debate on the Zulip before. I think my reasoning was something along the lines of, "If there's a typeclass `[p.prime]` argument for `padic_val_nat`, the usage is often the same and in cases where you make a mistake and `p` isn't prime, it sometimes helps you catch it at the beginning of your proof rather than the end". That being said, I agree the drawback of not being able to prove and use results where `p` is not prime is a big concern, and since most everyone.

I think I was trying here more to pitch a renaming. I think I would like something like `multiplicity'` better than `padic_val_nat` just because I feel like that would give me personally a better idea of the definition on non-prime values. To take a queue from your analogy to real division by zero in mathlib, I would probably be more likely to expect `padic_val_nat` to be 0 for not prime values of `p`, and `multiplicity'` I would expect to be nonzero.

But on the other hand, there is the consideration of tending to keep the status quo and also, maybe other people see `padic_val_nat` behave exactly as they expect on non-prime values. I'd be interested in your all's thoughts on this.